main.js

```diff
@@ -253,6 +254,21 @@ const regExGlobalNew = /script\.js\.global\./;
 function checkIsGlobal(obj) {
     return obj && obj.common && (regExGlobalOld.test(obj.common.name) || regExGlobalNew.test(obj._id));
 }
+
+function convertBackStringifiedValues(id, state) {
+    if (state && state.val && typeof state.val === 'string' &&
+        context.objects[id] && context.objects[id].common && context.objects[id].common.type &&
+        (context.objects[id].common.type === 'array' || context.objects[id].common.type === 'object')) {
```

Comment:
```suggestion
        context.objects[id] && context.objects[id].common &&
        (context.objects[id].common.type === 'array' || context.objects[id].common.type === 'object')) {
```
Also not necessary, it gets checked afterwards.